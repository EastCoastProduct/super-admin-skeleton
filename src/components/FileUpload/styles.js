import { StyleSheet } from 'aphrodite/no-important';
import { colors } from '../../styles/variables';
import { button } from '../../styles/mixins';

export default StyleSheet.create({
  holder: {
    marginBottom: 35,
  },
  file: {
    display: 'none',
  },
  emptyImage: {
    backgroundColor: colors.background,
    borderRadius: '50%',
    height: 200,
    padding: '75px 72px',
    position: 'relative',
    width: 200,
  },
  validatedHolder: {
    marginBottom: 50,
  },
  errHolder: {
    marginBottom: 15,
  },
  icon: {
    color: colors.header,
    fontSize: 50,
  },
  button: {
    ...button,
    bottom: 0,
    left: 0,
    position: 'absolute',
    textAlign: 'center',
    width: 150,
  },
  error: {
    margin: '5px 0 0',
  },
});
