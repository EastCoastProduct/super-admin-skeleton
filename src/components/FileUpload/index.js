import React, { PropTypes } from 'react';
import { css } from 'aphrodite/no-important';
import ErrorMsg from '../ErrorMsg';
import styles from './styles';

export const backgroundStyle = (value) => {
  if (!value) return {};
  const url = value instanceof FileList ? URL.createObjectURL(value[0]) : value;
  return {
    background: `url(${url}) center/cover no-repeat`,
  };
};

const FileUpload = (props) => {
  const { id, image, input: { value, ...input }, meta: { error }, preview,
    validated } = props;

  return (
    <p>
      <div
        className={css(styles.emptyImage, validated && styles.validatedImage,
          error && styles.errImage)}
        style={backgroundStyle(value || image)}
      >
        {!value && !image &&
          <i className={`${css(styles.icon)} fa fa-picture-o`} />
        }
        {!preview &&
          <label className={css(styles.button)} htmlFor={`${id}-${input.name}`}>
            Upload Photo
            <input
              className={css(styles.file)}
              id={`${id}-${input.name}`}
              type="file"
              {...input}
            />
          </label>
        }
      </div>
      {error && <ErrorMsg htmlFor={`${id}-${input.name}`}>{error}</ErrorMsg>}
    </p>
  );
};

FileUpload.defaultProps = {
  input: {},
  meta: {},
};

FileUpload.propTypes = {
  id: PropTypes.string,
  image: PropTypes.string,
  input: PropTypes.object,
  meta: PropTypes.object,
  preview: PropTypes.bool,
  validated: PropTypes.bool,
};

export default FileUpload;
