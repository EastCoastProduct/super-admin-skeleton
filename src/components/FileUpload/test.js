import React from 'react';
import { shallow } from 'enzyme';
import deepMerge from 'deepmerge';
import { createFile, createFileList } from '../../fixtures/fileAPI';
import FileUpload, { backgroundStyle } from './';

describe('FileUpload component snapshot', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });
  afterAll(() => {
    window.URL = undefined; // delete window.prop doesn't work
  });

  window.URL = {
    createObjectURL: jest.fn(() => 'ecp-logo-local-url'),
  };
  const image = 'link-to-image';
  const props = {
    id: 'EditUser',
    input: { name: 'image' },
    validated: true,
  };

  it('renders preview only', () => {
    const wrapper = shallow(<FileUpload image={image} preview={true} />);

    expect(wrapper).toMatchSnapshot();
    expect(backgroundStyle(image)).toEqual({
      background: `url(${image}) center/cover no-repeat`,
    });
  });

  it('renders empty validated file upload', () => {
    const wrapper = shallow(
      <FileUpload {...props} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(backgroundStyle()).toEqual({});
  });

  it('renders validate file upload with selected image', () => {
    const fileList = createFileList();
    const newProps = deepMerge(props, {
      input: {
        value: fileList,
      },
    });
    const wrapper = shallow(
      <FileUpload {...newProps} />
    );

    expect(window.URL.createObjectURL).toHaveBeenCalledWith(fileList[0]);
    expect(wrapper).toMatchSnapshot();
    expect(backgroundStyle(fileList)).toEqual({
      background: `url(undefined) center/cover no-repeat`,
    });
  });

  it('renders file upload with selected image with validation error', () => {
    const fileList = createFileList([createFile(2 * 1024 * 1024)]);
    const newProps = deepMerge(props, {
      input: {
        value: fileList,
      },
      meta: {
        error: 'Max. file size exceeded.',
      },
    });
    const wrapper = shallow(
      <FileUpload {...newProps} />
    );

    expect(window.URL.createObjectURL).toHaveBeenCalledWith(fileList[0]);
    expect(wrapper).toMatchSnapshot();
    expect(backgroundStyle(fileList)).toEqual({
      background: `url(undefined) center/cover no-repeat`,
    });
  });

  it('renders validate file upload with image prepopulated from S3', () => {
    const newProps = deepMerge(props, {
      image,
    });
    const wrapper = shallow(
      <FileUpload {...newProps} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(backgroundStyle(image)).toEqual({
      background: `url(${image}) center/cover no-repeat`,
    });
  });
});
