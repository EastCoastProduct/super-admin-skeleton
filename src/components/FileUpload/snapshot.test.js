import React from 'react';
import renderer from 'react-test-renderer';
import { FileList, File } from 'file-api';
import { StyleSheetTestUtils } from 'aphrodite/no-important';
import FileUpload from './';

describe('FileUpload component snapshot', () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    delete global.URL;
  });

  it('renders preview only', () => {
    const tree = renderer.create(
      <FileUpload image="link-to-image" preview={true} />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders empty validated file upload', () => {
    const tree = renderer.create(
      <FileUpload
        input={{ name: 'image' }}
        validated={true}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders validate file upload with selected image', () => {
    global.URL = { createObjectURL: jest.fn(() => 'ecp-logo-local-url') };
    const file = new FileList(new File('./ecp-logo.png'));
    const tree = renderer.create(
      <FileUpload
        input={{ name: 'image', value: file }}
        validated={true}
      />
    );

    expect(global.URL.createObjectURL).toHaveBeenCalledWith(file[0]);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders file upload with selected image with validation error', () => {
    global.URL = { createObjectURL: jest.fn(() => 'ecp-logo-local-url') };
    const file = new FileList(new File('./ecp-logo.png'));
    const tree = renderer.create(
      <FileUpload
        input={{ name: 'image', value: file }}
        meta={{ error: 'Max. file size exceeded.' }}
      />
    );

    expect(global.URL.createObjectURL).toHaveBeenCalledWith(file[0]);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders validate file upload with image prepopulated from S3', () => {
    const tree = renderer.create(
      <FileUpload
        image="link-to-image"
        input={{ name: 'image' }}
        validated={true}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
