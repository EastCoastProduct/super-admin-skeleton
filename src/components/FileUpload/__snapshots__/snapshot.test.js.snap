exports[`FileUpload component snapshot renders empty validated file upload 1`] = `
<div>
  <div
    className="emptyImage_avz902-o_O-validatedImage_4hnf83"
    style={Object {}}>
    <i
      className="icon_17id2n2 fa fa-picture-o" />
    <label
      className="button_h1ys7z"
      htmlFor="Form-image">
      Upload Photo
      <input
        className="file_avg348"
        id="Form-image"
        name="image"
        type="file" />
    </label>
  </div>
</div>
`;

exports[`FileUpload component snapshot renders file upload with selected image with validation error 1`] = `
<div>
  <div
    className="emptyImage_avz902-o_O-errImage_229bd5"
    style={
      Object {
        "background": "url(ecp-logo-local-url) center/cover no-repeat",
      }
    }>
    <label
      className="button_h1ys7z"
      htmlFor="Form-image">
      Upload Photo
      <input
        className="file_avg348"
        id="Form-image"
        name="image"
        type="file" />
    </label>
  </div>
  <p
    className="error_8cisgn">
    <i
      className="icon_1w16ibx fa fa-exclamation-triangle" />
    Max. file size exceeded.
  </p>
</div>
`;

exports[`FileUpload component snapshot renders preview only 1`] = `
<div>
  <div
    className="emptyImage_avz902"
    style={
      Object {
        "background": "url(link-to-image) center/cover no-repeat",
      }
    } />
</div>
`;

exports[`FileUpload component snapshot renders validate file upload with image prepopulated from S3 1`] = `
<div>
  <div
    className="emptyImage_avz902-o_O-validatedImage_4hnf83"
    style={
      Object {
        "background": "url(link-to-image) center/cover no-repeat",
      }
    }>
    <label
      className="button_h1ys7z"
      htmlFor="Form-image">
      Upload Photo
      <input
        className="file_avg348"
        id="Form-image"
        name="image"
        type="file" />
    </label>
  </div>
</div>
`;

exports[`FileUpload component snapshot renders validate file upload with selected image 1`] = `
<div>
  <div
    className="emptyImage_avz902-o_O-validatedImage_4hnf83"
    style={
      Object {
        "background": "url(ecp-logo-local-url) center/cover no-repeat",
      }
    }>
    <label
      className="button_h1ys7z"
      htmlFor="Form-image">
      Upload Photo
      <input
        className="file_avg348"
        id="Form-image"
        name="image"
        type="file" />
    </label>
  </div>
</div>
`;
